db/view/view.cc

```diff
@@ -1777,8 +1777,10 @@ future<> view_update_generator::mutate_MV(
                     mut.s->ks_name(), mut.s->cf_name(), base_token, view_token);
             local_view_update = _proxy.local().mutate_mv_locally(mut.s, *mut_ptr, tr_state, db::commitlog::force_sync::no).then_wrapped(
                     [s = mut.s, &stats, &cf_stats, tr_state, base_token, view_token, my_address, mut_ptr = std::move(mut_ptr),
-                            sem_units] (future<>&& f) {
+                            sem_units, this] (future<>&& f) mutable {
                 --stats.writes;
+                sem_units = nullptr;
```

Comment:
why is zeroing sem_units a few lines earlier important? Is it necessary for this patch, or just an improvement you thought of while doing this?

Reply:
I want it to be zeroed before we update the view update backlog, otherwise we'll always be updating the backlog with a  value that's just about to change